### ISSUE TITLE
🐛 Astro dynamic redirects fail `biome(parse)`

### ISSUE THREAD

**Issue by AdoryVo** [biomejs/biome#2232]
### Environment information

```block
CLI:
  Version:                      1.6.3
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           "v20.10.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/8.15.5"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

1. Add an [Astro dynamic redirect](https://docs.astro.build/en/guides/routing/#dynamic-redirects) to a `.astro` file. This involves adding `return Astro.redirect('/url')` to your Astro [component script](https://docs.astro.build/en/basics/astro-components/#the-component-script).
2. Observe "Illegal return statement outside of a function" from biome(parse) in syntax highlighting as well as `biome check` output.  See [Biome playground snippet](https://biomejs.dev/playground/?code=LQAtAC0ACgBjAG8AbgBzAHQAIABsAG8AZwBnAGUAZABJAG4AIAA9ACAAZgBhAGwAcwBlADsACgAKAC8ALwAgAEkAZgAgAHQAaABlACAAdQBzAGUAcgAgAGkAcwAgAG4AbwB0ACAAbABvAGcAZwBlAGQAIABpAG4ALAAgAHIAZQBkAGkAcgBlAGMAdAAgAHQAaABlAG0AIAB0AG8AIAB0AGgAZQAgAGwAbwBnAGkAbgAgAHAAYQBnAGUACgBpAGYAIAAoACEAbABvAGcAZwBlAGQASQBuACkAIAB7AAoAIAAgAHIAZQB0AHUAcgBuACAAQQBzAHQAcgBvAC4AcgBlAGQAaQByAGUAYwB0ACgAJwAvAGwAbwBnAGkAbgAnACkAOwAKAH0ACgAtAC0ALQAKADwAaAB0AG0AbAA%2BAAoAIAAgADwAIQAtAC0AIABQAGEAZwBlACAAaABlAHIAZQAuAC4ALgAgAC0ALQA%2BAAoAPAAvAGgAdABtAGwAPgA%3D) and picture.

![image](https://github.com/biomejs/biome/assets/33337855/46c75bcf-e947-4596-9f1b-1c8d447c0159)

### Expected result

The return statement is legal in Astro files, as shown in the Astro docs, so an error should not be thrown.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by erguotou520** [comment]
How to disable this rule?

**Comment by Sec-ant** [comment]
> How to disable this rule?

This is a parser error so it cannot be disabled. This issue should be fixed in #2273.

**Comment by winixt** [comment]
> > How to disable this rule?
> 
> This is a parser error so it cannot be disabled. This issue should be fixed in [#2273](https://github.com/biomejs/biome/pull/2273).

oxc can control by `allowReturnOutsideFunction` config
